pkg/storage/provider_abort_action_test.go

```diff
@@ -0,0 +1,272 @@
+package storage_test
+
+import (
+	"testing"
+
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/fixtures"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/internal/fixtures/testusers"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/storage/internal/testabilities"
+	"github.com/bsv-blockchain/go-wallet-toolbox/pkg/wdk"
+	"github.com/go-softwarelab/common/pkg/to"
+	"github.com/stretchr/testify/require"
+)
+
+const NumberOfDesiredUTXOs = 32
+
+func TestAbortActionSuccess(t *testing.T) {
+	// given:
+	given, cleanup := testabilities.GivenCustomStorage(t, fixtures.StorageServerPrivKey, "dbstorage_test")
+	defer cleanup()
+
+	activeStorage := given.Provider().GORM()
+	faucet := given.Faucet(activeStorage, testusers.Alice)
+	faucet.TopUp(100_000)
+
+	createResult, err := activeStorage.CreateAction(
+		t.Context(),
+		testusers.Alice.AuthID(),
+		fixtures.DefaultValidCreateActionArgs(),
+	)
+	require.NoError(t, err)
+	testabilities.ThenDBState(t, activeStorage).AllOutputs(testusers.Alice).WithCount(1 + NumberOfDesiredUTXOs)
+	// when:
+	result, err := activeStorage.AbortAction(
+		t.Context(),
+		testusers.Alice.AuthID(),
+		wdk.AbortActionArgs{
+			Reference: to.Ptr(createResult.Reference),
+		},
+	)
+
+	// then:
+	require.NoError(t, err)
+	require.NotNil(t, result)
+	require.True(t, result.Aborted)
+
+	thenDBState := testabilities.ThenDBState(t, activeStorage)
+	thenDBState.HasUserTransactionByReference(testusers.Alice, createResult.Reference).
+		WithStatus(wdk.TxStatusFailed)
+	thenDBState.AllOutputs(testusers.Alice).WithCount(1)
+}
+
+func TestAbortActionTransactionNotOutgoing(t *testing.T) {
+	// given:
+	given, cleanup := testabilities.GivenCustomStorage(t, fixtures.StorageServerPrivKey, "dbstorage_test")
+
+	defer cleanup()
+
+	activeStorage := given.Provider().GORM()
+
+	given.Faucet(activeStorage, testusers.Alice).TopUp(100_000)
+	createdTx, _ := given.ActionCreatedAndSigned(activeStorage)
```

Comment:
Interesting - I would assume that result of this would be exactly the outgoing action, 
Maybe it's worth checking if this is not a bug.

@chris-4chain Was this intentional or we should create an issue to investigate

Reply:
@chris-4chain nvm

Ohh, I see that I overlooked next line. So you just taking the parent tx.

I'm suggesting to either 
 use the txID from faucet topup
OR add some better spacing to better expose the line from where the txID is comming from.


Additionally, it seems that you can use this `given.ActionCreatedAndSigned(activeStorage)` in all the other tests, instead of plain call to CreateAction